task_02/tevfik_aksoy/main.cpp

```diff
@@ -0,0 +1,183 @@
+#include <fstream>
+
+using std::cout;
+using std::endl;
+using std::get;
+using std::to_string;
+using std::vector;
+
+//Tuple of <int,int> for realization of vector of vectors of <int,int>
+typedef std::tuple<int, int> DestID_EdgeID;
+
+//Structure of nodes(vertices)
+struct Node
+{
+private:
+	int id = 0;
+
+public:
+	explicit Node(int id) : id(id) {}
+	explicit Node() : id(0) {}
+
+	const int GetID() const
+	{
+		return id;
+	}
+
+	void SetID(int val)
+	{
+		id = val;
+	}
+
+	std::string to_json(vector<int> edge_ids, bool end_flag) const
+	{
+		std::string str = "\t{ \"id\": " + to_string(id) + ", \"edge_ids\": [";
+		for (int i = 0; i < edge_ids.size(); i++)
+		{
+			str += to_string(edge_ids[i]);
+			if (i + 1 == edge_ids.size())
+			{
+				str += "] }";
+				if (!end_flag)
+					str += ",\n";
+				else
+					str += "\n  ],\n";
+			}
+			else
+				str += ", ";
+		}
+		return str;
+	}
+};
+
+//Structure of the edge including the source and destination nodes in it
+struct Edge
+{
+private:
+	int id = 0;
+
+public:
+	Node src_node;
+	Node dest_node;
+	Edge(int src_id, int dest_id, int id)
+	{
+		src_node.SetID(src_id);
+		dest_node.SetID(dest_id);
+		this->id = id;
+	}
+
+	Edge() : id(0), src_node(Node()), dest_node(Node()) {}
+
+	const int getID() const
+	{
+		return id;
+	}
+
+	void SetID(int val)
+	{
+		id = val;
+	}
+
+	std::string to_json(int end_flag) const
+	{
+		std::string str = "\t{ \"id\": " + to_string(id) + ", \"vertex_ids\": [" + to_string(src_node.GetID()) + ", " + to_string(dest_node.GetID()) + "] }";
+		if (!end_flag)
+			str += ",\n";
+		else
+			str += "\n  ]\n}\n";
+		return str;
+	}
+};
+
+class Graph
+{
+private:
+	int total_nodes_;
+	int total_edges_;
+	vector<std::pair<int, int>> edgeSrcDest; //vector of pairs to keep data of source and destination vertices of an edge
+	vector<vector<DestID_EdgeID>> graphVec;
+
+public:
+	Graph()
+	{
+		generateVertices();
+	}
+
+	void generateVertices()
+	{
+		const vector<Edge> edges =
+			{{0, 1, 0}, {0, 2, 1}, {0, 3, 2}, {1, 4, 3}, {1, 5, 4}, {1, 6, 5}, {2, 7, 6}, {2, 8, 7}, {3, 9, 8}, {4, 10, 9}, {5, 10, 10}, {6, 10, 11}, {7, 11, 12}, {8, 11, 13}, {9, 12, 14}, {10, 13, 15}, {11, 13, 16}, {12, 13, 17}};
+		//using set to exclude the repeating source nodes(the ones with more than 1 edge connected)
+		std::set<int> nodes_set;
+		for (int i = 0; i < edges.size(); i++)
+		{
+			nodes_set.insert(edges[i].src_node.GetID());
+		}
+		total_nodes_ = nodes_set.size();
+		total_edges_ = edges.size();
+		graphVec.resize(total_nodes_ * 2);
+		for (const auto &edge : edges)
+		{
+			const int src_id = edge.src_node.GetID();
+			if (src_id >= total_nodes_)
+				continue;
+			const int dest_id = edge.dest_node.GetID();
+			const int edge_id = edge.getID();
+
+			edgeSrcDest.push_back(std::make_pair(src_id, dest_id));
+
+			graphVec[src_id].push_back(std::make_tuple(dest_id, edge_id));
+			graphVec[dest_id].push_back(std::make_tuple(src_id, edge_id));
+		}
+	}
+
+	void to_json() const

```

Comment:
Работа с файлами не входит в зону отвественности графа.
Эта функция скорее должна возвращать строку, которая извне уже может быть записана в файл.
```suggestion
      std::string to_json() const {
```